test/cmocka/include/sqrt_tables.h

```diff
@@ -0,0 +1,74 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2021 Intel Corporation. All rights reserved.
+ *
+ * Author: Shriram Shastry <malladi.sastry@linux.intel.com>
+ */
+/* Include Files */
+#include <stdint.h>
+
+/* Reference table generated by Matlab sqrt() */
+static const double sqrt_ref_table[] = {
```

Comment:
why is this in a header? you only use it in one file.

Reply:
yes, I kept purposely to have reference table for all Math library function [ `trig_tables.h` ]. So following as a rule . Future `sqrt_int32()` can be reuse with new reference table with higher accuracy.

Reply:
Its not duplication . current `sqrt_int16()` is using `sqrt_ref_table` in `square_root.c`.

For future `sqrt_int32()` a different table is  required for evaluating error. 
In that case, `sqrt_ref_table_int32` table will be added to `sqrt_tables.h `file  followed  with new `square_root_int32.c` file . `square_root_int32.c` file include `sqrt_tables.h` same header  file  for  evaluation  of new function `sqrt_int32()`. 
Above description is followed for cordic/trigonometric functions.

Reply:
That isn't what Marc means. C when you include a header it explicitly copies the content of the header into the C file. Therefore if it is included more than once it will duplicate it, so it is better to not put this here if we dont need to.

Reply:
Understood !! OK, I will move reference `sqrt_ref_table` to `square_root.c`

Reply:
> blow up memory usage

Read too fast and didn't realize this is only for unit tests. So memory usage does not matter much.

> like trig_tables.h 

Interesting example, thanks:
```shell

cd sof/build_ut_defs/apollolake_defconfig/test/cmocka/src/math/trig/CMakeFiles


for o in */*_fixed.c.o; do echo " === $o ==="; nm $o | grep _ref_table; done

 === acos_16b_fixed.dir/acos_16b_fixed.c.o ===
0000000000000680 r acos_ref_table
 === acos_32b_fixed.dir/acos_32b_fixed.c.o ===
0000000000000680 r acos_ref_table
 === asin_16b_fixed.dir/asin_16b_fixed.c.o ===
0000000000000680 r asin_ref_table
 === asin_32b_fixed.dir/asin_32b_fixed.c.o ===
0000000000000680 r asin_ref_table
 === cos_16b_fixed.dir/cos_16b_fixed.c.o ===
0000000000000020 r cos_ref_table
 === cos_32b_fixed.dir/cos_32b_fixed.c.o ===
0000000000000020 r cos_ref_table
 === sin_16b_fixed.dir/sin_16b_fixed.c.o ===
0000000000000020 r sin_ref_table
 === sin_32b_fixed.dir/sin_32b_fixed.c.o ===
0000000000000020 r sin_ref_table
```

So it looks like the `*_ref_table`s are duplicated. But in fact these `.o` files are not linked together, they're only used one at a time. So not really. But this is confusing.

In general .h files are for declarations and .c files for definitions. There are some exceptions but I still don't think one exception is required here. I think you could leave only the declarations in the `trig_tables.h` file:

```C
static const float sin_ref_table[];
static const float cos_ref_table[];
...
```

and then create a new, `common_ref_tables.c` with all the definitions and link every test to it. This would look like regular C code and be much easier to read and review because it would not raise questions.